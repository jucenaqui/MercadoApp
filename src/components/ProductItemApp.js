import React from 'react';

export const ProductItemApp = ( { product}) => {
    
    return (
        <>
      
       <div>
                    <img src={product.thumbnail}  alt={product.title}/>
                    <p>{product.title}</p>
                    <h3>{product.price}</h3>
                </div>
        </>
    )
}
