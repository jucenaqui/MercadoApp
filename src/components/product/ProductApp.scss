.product-list {
    background-color: white;
    padding: 10px;
    display: flex;
    flex-direction: column;
    justify-content: center;
    width: 80%;
    margin: 20px auto;
}