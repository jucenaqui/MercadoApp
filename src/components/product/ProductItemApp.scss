.product-item {
    display: flex;
    flex-wrap: wrap;
    justify-content: flex-start;
    padding: 10px 20px;
    box-shadow: 0px 1px 5px #f3f2f2;

    img {
        margin: 10px;
        border-radius: 3px;
        width: 100px;
    }
}

.product-item-description {
    align-items: flex-start;
    box-sizing: border-box;
    display: flex;
    flex-direction: column;
    flex-wrap: nowrap;
    padding-top: 20px;

    p {
        margin: 0 !important;
        width: 90%;
    }

    h3 {
        margin: 10px 0;
    }
}

@media screen and (max-width: '768px') {
    .product-item {
        flex-direction: column;
        justify-content: center;
        align-items: center;

        img {
            width: 150px;
        }
    }

    .product-item-description {
        align-items: center;

        p {
            width: 100%;
            padding: 5px 20px;
        }
    }
}