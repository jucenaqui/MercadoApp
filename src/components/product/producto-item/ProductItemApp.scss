.product-item-app {
    display: flex;
    flex-wrap: wrap;
    justify-content: flex-start;
    padding: 10px 20px;
    box-shadow: 0px 1px 5px #f3f2f2;
    cursor: pointer;

    img {
        margin: 10px;
        border-radius: 3px;
        width: 100px;
    }
}

.product-description-app {
    align-items: flex-start;
    box-sizing: border-box;
    display: flex;
    flex-direction: column;
    flex-wrap: nowrap;
    padding-top: 20px;

    p {
        margin: 0 !important;
        width: 90%;
    }

    h3 {
        margin: 10px 0;
    }
}

@media screen and (max-width: '768px') {
    .product-item-app {
        flex-direction: column;
        justify-content: center;
        align-items: center;

        img {
            width: 150px;
        }
    }

    .product-description-app {
        align-items: center;

        p {
            width: 100%;
            padding: 5px 20px;
            text-align: center;
        }
    }
}