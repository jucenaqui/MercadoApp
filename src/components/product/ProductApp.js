import React from 'react';
import {ProductItemApp} from './ProductItemApp';

import './ProductApp.scss';

const ProductApp = ( { products } ) => {

    return (
     <>
        <ol className="product-list">
            {
                products.map((product)=> (<ProductItemApp key={product.id} product={product} />))
            }
        </ol>
     </>
    )
}

export default ProductApp;