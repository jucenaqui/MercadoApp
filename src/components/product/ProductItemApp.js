import React from 'react';

import './ProductItemApp.scss';

export const ProductItemApp = ( { product}) => {
    
    return (
        <>
      
            <div className="product-item">
                <img className="producto-item-img" src={product.thumbnail}  alt={product.title}/>
                <div className="product-item-description"> 
                <h3>$ {product.price}</h3>   
                <p>{product.title}</p>
                </div>
            </div>
        </>
    )
}
