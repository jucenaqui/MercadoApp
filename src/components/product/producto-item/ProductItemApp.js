import React from 'react';
import { useHistory } from "react-router-dom";

import PropTypes from 'prop-types';

import './ProductItemApp.scss';

export const ProductItemApp = ( { product} ) => {

    const history = useHistory();

    const handleLink = () => {
        history.push(`items/${product.id}`);
    }
    
    return (
        <>
            <div className="product-item-app" onClick={ handleLink }>
                <img className="producto-item-img-app" src={product.picture}  alt={product.title}/>
                <div className="product-description-app"> 
                    <h3>$ {product.price?.amount}</h3>   
                    <p>{product.title}</p>
                </div>
            </div>
        </>
    )
}

ProductItemApp.prototype = {
    product: PropTypes.object.isRequired
}