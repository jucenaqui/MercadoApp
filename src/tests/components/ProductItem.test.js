import { shallow } from "enzyme"
import { ProductItemApp } from "../../components/product/producto-item/ProductItemApp"
import { productMuck } from '../mocks/product';

describe('Pruebas <ProductItem />', () => {
    
    const product = productMuck;
    const wrapper = shallow(<ProductItemApp product={ product} />);

    test('should show the component succesfull', () => {
        expect( wrapper ).toMatchSnapshot();
    });

    test('should have a <p> with the title', () => { 
        const title = 'Heladera';
        const p = wrapper.find('p');
        expect(p.text().trim()).toBe(title);
    });

    test('should have a <h3> with a price', () => {
        const price = `$ ${product.price.amount}`;
        const h3 = wrapper.find('h3');
        expect(h3.text().trim()).toBe(price);
    });
    
    test('should have a img with a src and alt attibutes', () => {
        const imgSrc = product.picture;
        const imgAlt = product.title;
        const img = wrapper.find('img');
        expect(img.prop('src')).toBe(imgSrc)
        expect(img.prop('alt')).toBe(imgAlt);
        expect(img.prop('src').includes('http://')).toBe(true);
    });
    

})
