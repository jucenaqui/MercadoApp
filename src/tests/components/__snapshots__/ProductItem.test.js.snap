// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas <ProductItem /> should show the component succesfull 1`] = `
<Fragment>
  <div
    className="product-item-app"
    onClick={[Function]}
  >
    <img
      alt="Heladera"
      className="producto-item-img-app"
      src="http://http2.mlstatic.com/D_965028-MLA32542008594_102019-I.jpg"
    />
    <div
      className="product-description-app"
    >
      <h3>
        $ 
        47200
      </h3>
      <p>
        Heladera
      </p>
    </div>
  </div>
</Fragment>
`;
